app/src/main/java/com/crossbowffs/nekosms/xposed/SmsHandlerHook.java

```diff
@@ -187,7 +187,9 @@ private void afterConstructorHandler(XC_MethodHook.MethodHookParam param) {
                 PreferenceConsts.REMOTE_PREFS_AUTHORITY,
                 PreferenceConsts.FILE_MAIN,
                 true);
-            grantWriteSmsPermissions(context);
```

Comment:
Have you tested that the app still works without this? You will need to uninstall the app to verify, as this write is persistent.